controllers/metric_controller.go

```diff
@@ -250,21 +260,11 @@ func (r *metricReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-// isMonitoringEnabled returns true if the given object namespace has monitoring label set to true
-func isMonitoringEnabled(obj runtime.Object, watchNamespace string) bool {
+// isOperatorNamespace returns true if the given object is the operator namespace
+func isOperatorNamespace(obj runtime.Object, watchNamespace string) bool {
 	namespace, ok := obj.(*v1.Namespace)
 	if !ok {
 		return false
 	}
-	if namespace.GetName() != watchNamespace {
-		return false
-	}
-	if value, ok := namespace.Labels["openshift.io/cluster-monitoring"]; ok {
```

Comment:
I'm not following why the label check is no longer being made

Reply:
@sebsoto label check is being made in the reconciler, so we can generate events if the label has not been added to the namespace.

Reply:
I guess I am not understanding the issue that is being solved.
It seems like the underlying bug is complaining about `event` generation indicating that monitoring is being disabled/enabled.

Reply:
So in possibly in addition to this work, you would want to keep track of the current state, and if it changes, generate an event indicating it has changed. Right now an event is generated indicating it is not enabled.
https://github.com/openshift/windows-machine-config-operator/blob/641adab042a1c97b18832d3fb02da176e60b91f4/controllers/metric_controller.go#L93-L110


The followup question is, is this something that we want to do. I see some value in it.

Reply:
Correct that's the bug, this event was generated before the controller changes went in. Agreed, I see value in generating an event when there is a change.